src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java

```diff
@@ -21,7 +26,23 @@
 public class VulnerabilityTrendHelper {
 
     public static ContrastSDK createSDK(String username, String serviceKey, String apiKey, String teamServerUrl) {
```

Comment:
Some thoughts on this method:

- I think the `catch MalformedURLException` is catching a badly-formed teamserverUrl, no? If so, let's move that out of the if block at put it at the top of the method with a logged message. After all, if the teamserverUrl is bad, then we can't do anything

- That will clean up the if statement and bit and turn it into

```
if(proxyConfig != null) {
  contrastSDK = new ContrastSDK(username, serviceKey, apiKey, teamServerUrl, Proxy.NO_PROXY);
} else {
  Proxy proxy = proxyConfig.createProxy(new URL(teamServerUrl).getHost());
  contrastSDK = new ContrastSDK(username, serviceKey, apiKey, teamServerUrl, proxy);
}
```

And it'll allow us to avoid the check on line 41

Reply:
The try-catch block is now moved to the top of the function, but we still need to check if sdk is null in the end. Otherwise we would need to duplicate the contrastSDK assignment for each of the above"if" blocks.